src/main/java/boost/project/LibertyServerConfigGenerator.java

```diff
@@ -0,0 +1,153 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package boost.project;
+
+import java.io.File;
+import java.util.List;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+
+/**
+ * Create a Liberty server.xml
+ *
+ */
+public class LibertyServerConfigGenerator {
+
+	Document doc;
+	Element featureManager;
+	Element httpEndpoint;
+
+	
+	public LibertyServerConfigGenerator() {
+	
+		try {
+
+			DocumentBuilderFactory docFactory = DocumentBuilderFactory.newInstance();
+			DocumentBuilder docBuilder = docFactory.newDocumentBuilder();
+
+			// Create top level server config element
+			doc = docBuilder.newDocument();
+			Element serverRoot = doc.createElement("server");
+			serverRoot.setAttribute("description", "Liberty server generated by Liberty Boost");
+			doc.appendChild(serverRoot);
+			
+			// Create featureManager config element
+			featureManager = doc.createElement("featureManager");
+			serverRoot.appendChild(featureManager);
+			
+			// Create httpEndpoint config element with default values
+			httpEndpoint = doc.createElement("httpEndpoint");
+			httpEndpoint.setAttribute("id", "defaultHttpEndpoint");
+			httpEndpoint.setAttribute("httpPort", "9080");
+			httpEndpoint.setAttribute("httpsPort", "9443");
+			serverRoot.appendChild(httpEndpoint);
+			
+
+	  } catch (ParserConfigurationException pce) {
+		pce.printStackTrace();
+	  }
+	}
+	
+	/**
+     * Set HTTP endpoint configuration for this server config instance.
+     *
+     * @param host The host to use for this server. 
+     * @param httpPort The HTTP port to use for this server. 
+     * @param httpsPort The HTTPS port to use for this server.
+     */
+	public void setHttpEnpoints(String host, String httpPort, String httpsPort) {
+		
+		if (host != null) {
+			httpEndpoint.setAttribute("host", host);
+		}
+		if (httpPort != null) {
+			httpEndpoint.setAttribute("httpPort", httpPort);
+		}
+		if (httpsPort != null) {
+			httpEndpoint.setAttribute("httpsPort", httpsPort);
+		}
+	}
+	
+	/**
+	 * Add a Liberty feature to the server configuration
+	 *
+	 * @param featureName The full name of the Liberty feature to add. 
+     */
+	public void addFeature(String featureName) {
+
+		boolean featureAlreadyExists = false;
+		
+		NodeList existingFeatures = featureManager.getChildNodes();
```

Comment:
Might be easier to keep a Set<String> of feature names that have already been added. Then you could do a simple check if the Set.contains(featureName) before adding the element.

Reply:
Good idea.